pybossa/sched.py

```diff
@@ -300,16 +347,105 @@ def template_get_locked_task(project_id, user_id=None, user_ip=None,
             timeout = timeout or TIMEOUT
             remaining = float('inf') if calibration else n_answers - taskcount
             if acquire_lock(task_id, user_id, remaining, timeout):
+                # reserve tasks
+                acquire_reserve_task_lock(project_id, task_id, user_id, timeout)
```

Comment:
Should we check whether the function `acquire_reserve_task_lock` returns success?

Reply:
At 351, user has already obtained lock on the task, hence I think it is desired to return task to the user. In worst case if user don't acquire reserve task lock, there is likelihood of obtaining next task from non reserve task category ... in a rare case.